src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -233,7 +247,10 @@ public async Task<IActionResult> Index(BulkActionViewModel model)
                 return Forbid();
             }
 
-            var allSettings = _shellHost.GetAllSettings();
+            if (model.BulkAction.ToString() == "Remove" && !_tenantsOptions.TenantRemovalAllowed)
```

Comment:
```suggestion
            if (model.BulkAction.ToString() == nameof(TenantsBulkAction.Remove) && !_tenantsOptions.TenantRemovalAllowed)
```

Reply:
Idem, to not have to also change the existing ones just before and then add too much diff.

But okay, done on my side.